src/credential/trustmark/get-credential-trustmark.ts

```diff
@@ -0,0 +1,69 @@
+import { SignJWT, type CryptoContext } from "@pagopa/io-react-native-jwt";
+import { WalletInstanceAttestation } from "@pagopa/io-react-native-wallet";
+import { IoWalletError } from "../../utils/errors";
+import { obfuscateString } from "../../utils/string";
+
+export type GetCredentialTrustmarkJwt = (
+  walletInstanceAttestation: string,
+  wiaCryptoContext: CryptoContext,
+  credentialType: string,
+  documentNumber?: string
+) => Promise<string>;
+
+/**
+ * Generates a trustmark signed JWT, which is used to verify the authenticity of a credential.
+ *
+ * @param walletInstanceAttestation the Wallet Instance's attestation
+ * @param wiaCryptoContext The Wallet Instance's crypto context associated with the walletInstanceAttestation parameter
+ * @param credentialType The type of credential for which the trustmark is generated
+ * @param documentNumber (Optional) Document number contained in the credential, if applicable
+ * @returns A promise that resolves to the signed JWT string, representing the credential's trustmark.
+ */
+export const getCredentialTrustmarkJwt: GetCredentialTrustmarkJwt = async (
+  walletInstanceAttestation,
+  wiaCryptoContext,
+  credentialType,
+  documentNumber
+): Promise<string> => {
+  /**
+   * Check that the public key used to sign the trustmark is the one used for the WIA
+   */
+  const holderBindingKey = await wiaCryptoContext.getPublicKey();
+  const decodedWia = WalletInstanceAttestation.decode(
+    walletInstanceAttestation
+  );
+
+  if (
+    !decodedWia.payload.cnf.jwk.kid ||
+    decodedWia.payload.cnf.jwk.kid !== holderBindingKey.kid
+  ) {
```

Comment:
not for this PR, but usually a check on the kid alone might not be enough. The thumbprint of both keys should be calculated and compared.

Reply:
I've updated the check here: https://github.com/pagopa/io-react-native-wallet/pull/156/commits/88bd9ee07565b9bdd43b0b2a219ee74f0eb397de
Let me know!

Reply:
Top 🎆